com.unity.perception/Runtime/GroundTruth/Labelers/KeypointLabeler.cs

```diff
@@ -193,65 +287,181 @@ void OnInstanceSegmentationImageReadback(int frameCount, NativeArray<Color32> da
+                }
+
+                checkLocations.Dispose();
+            }
+        }
+
+        private void DoDepthCheck(ScriptableRenderContext scriptableRenderContext, List<KeypointEntry> keypointEntries, NativeList<float3> checkLocations)
+        {
+            var keypointCount = keypointEntries.Count * activeTemplate.keypoints.Length;
+
+            var commandBuffer = CommandBufferPool.Get("KeypointDepthCheck");
+
+            var textureDimensions = TextureDimensions(keypointCount);
+
+
+            SetupDepthCheckBuffers(checkLocations.Length);
+
+            var positionsPixeldata = new NativeArray<half>(textureDimensions.x * textureDimensions.y * 2, Allocator.Temp);
+            var depthPixeldata = new NativeArray<half>(textureDimensions.x * textureDimensions.y, Allocator.Temp);
+            var depthTexture = Shader.GetGlobalTexture("_CameraDepthTexture");
+
+            for (int i = 0; i < checkLocations.Length; i++)
+            {
+                var pos = checkLocations[i];
+                positionsPixeldata[i * 2] = new half(pos.x);
+                positionsPixeldata[i * 2 + 1] = new half(pos.y);
+                depthPixeldata[i] = new half(pos.z);
+            }
+
+            m_KeypointPositionsTexture.SetPixelData(positionsPixeldata, 0);
+            m_KeypointPositionsTexture.Apply();
+            m_KeypointCheckDepthTexture.SetPixelData(depthPixeldata, 0);
+            m_KeypointCheckDepthTexture.Apply();
+
+            positionsPixeldata.Dispose();
+            depthPixeldata.Dispose();
+
+            m_MaterialDepthCheck.SetTexture("_Positions", m_KeypointPositionsTexture);
+            m_MaterialDepthCheck.SetTexture("_KeypointCheckDepth", m_KeypointCheckDepthTexture);
+            m_MaterialDepthCheck.SetTexture("_CameraDepthTexture", depthTexture);
+            commandBuffer.Blit(null, m_ResultsBuffer, m_MaterialDepthCheck);
+
+            scriptableRenderContext.ExecuteCommandBuffer(commandBuffer);
+            scriptableRenderContext.Submit();
+            CommandBufferPool.Release(commandBuffer);
+            m_DepthCheckReader.Capture(scriptableRenderContext, OnDepthCheckReadback);
+        }
+
+        private static Vector2Int TextureDimensions(int keypointCount)
+        {
+            var width = Math.Max(k_MinTextureWidth, Mathf.NextPowerOfTwo((int)Math.Ceiling(Math.Sqrt(keypointCount))));
+            var height = width;
+
+            var textureDimensions = new Vector2Int(width, height);
+            return textureDimensions;
+        }
+
+        private void OnDepthCheckReadback(int frame, NativeArray<Color32> data, RenderTexture renderTexture)
+        {
+            DoDepthCheckReadback(frame, data);
+        }
+
+        private void OnDepthCheckReadback(int frameCount, AsyncGPUReadbackRequest obj)
+        {
+            var data = obj.GetData<uint>();
+            //DoDepthCheckReadback(frameCount, data);
+        }
```

Comment:
Is this call needed because the meat of this call appears to be commented out.

Reply:
Nah, deleted it.